File: master/master/page/company_info/company_info.py
import frappe
from frappe import _
from frappe.model.document import Document

# @frappe.whitelist()
# def get_count(Company,Plant,Location,Department,Spareslist,Machinelist):
#     Company_count = frappe.db.count(Company);
#     Plant_count = frappe.db.count(Plant);
#     Location_count = frappe.db.count(Location);
#     Department_count = frappe.db.count(Department);
#     Spare_count = frappe.db.count(Spareslist);
#     Machine_count = frappe.db.count(Machinelist);

#     return Company_count,Plant_count,Location_count,Department_count,Spare_count,Machine_count
    
# @frappe.whitelist()
# def get_doctype(Company):
#     doc = frappe.db.get_all('Company',fields=['company_code','company_name','company_email','company_address']);

#     return doc


@frappe.whitelist(allow_guest='True')
def get_company():
    company = frappe.get_all('Company', fields=['company_code','company_name','company_address','company_state','company_country','company_city','company_email','company_pincode','company_phone','company_mobile','company_website','company_gstinno','company_status'])
    ht = "<table>"
    ht+= "<tr>"
    
    for i in company[0].keys():
        ht += f'<th style="background-color:#558b21">{i}</th>'

    ht+= "</tr>"
    for j in company:
        ht += "<tr>"
        for i,k in j.items():
            ht += f'<td>{k}</td>'

        ht += "</tr>"
    

    ht += "</table>"
    return ht

import json
@frappe.whitelist()
def save_Company(data):
    data = json.loads(data)
    print(data)
    company_code = data['company_code']
    company_name = data['company_name']
    company_address = data['company_address']
    company_country = data['company_country']
    company_state = data['company_state']
    company_city = data['company_city']
    company_email = data['company_email']
    company_pincode = data['company_pincode']
    company_phone = data['company_phone']
    company_mobile = data['company_mobile']
    company_gstinno = data['company_gstinno']
    company_website = data['company_website']
    company_status = data['company_status']



    company = frappe.new_doc("Company")

    company.company_code = company_code
    company.company_name = company_name
    company.company_address = company_address
    company.company_country = company_country
    company.company_state = company_state
    company.company_city = company_city
    company.company_email = company_email
    company.company_pincode = company_pincode
    company.company_mobile = company_mobile
    company.company_phone = company_phone
    company.company_gstinno = company_gstinno
    company.company_website = company_website
    company.company_status = company_status
    
    company.insert(ignore_permissions=True)
    return "Company sucessfully submitted"

File: master/master/page/company_info/company_info.js
frappe.pages['company-info'].on_page_load = function(wrapper) {
    var page = frappe.ui.make_app_page({
        parent: wrapper,
        title: 'Company',
        single_column: true
    });

    // Ensure the template exists and is loaded correctly
    $(frappe.render_template("company_info", {})).appendTo(page.body);


    function navigateTo(route) {
        // Use Frappe's built-in routing function to navigate
        frappe.set_route(route);
    }

    // Event listener to handle clicks on navigation links
    document.querySelectorAll('.dropdown-content a').forEach(link => {
        link.addEventListener('click', function(event) {
            event.preventDefault(); // Prevent default link behavior
            navigateTo(this.getAttribute('data-route')); // Get route from data-route attribute and navigate
        });
    });

    
    frappe.call({
        method:'master.master.page.company_info.company_info.get_company',
        args: {},
        callback:function(response){
            var m = response.message;
            document.getElementById("company").innerHTML=m;
            console.log(m);
        }
    });

    // Handling form submission
    var companyform = document.querySelector("#companyform");
    if (companyform) {
        console.log(companyform);

        companyform.addEventListener("submit", function(event) {
            event.preventDefault();

            var formData = {
                
                'company_code': document.querySelector("#company_code").value,
                'company_name': document.querySelector("#company_name").value,
                'company_address': document.querySelector("#company_address").value,
                'company_country': document.querySelector("#company_country").value,
                'company_state': document.querySelector("#company_state").value,
                'company_city': document.querySelector("#company_city").value,
                'company_email': document.querySelector("#company_email").value,
                'company_pincode': document.querySelector("#company_pincode").value,
                'company_phone': document.querySelector("#company_phone").value,
                'company_mobile': document.querySelector("#company_mobile").value,
                'company_gstinno': document.querySelector("#company_gstinno").value,
                'company_website': document.querySelector("#company_website").value,
                'company_status': document.querySelector("#company_status").value
            };

            console.log("FormData", formData);

            frappe.call({
                method: 'master.master.page.company_info.company_info.save_Company',
                args: {
                    'data': formData
                },
                callback: function(response){
                    console.log(response);
                    if(response.message){
                        alert("You successfully submitted");
                    }
                    else {
                        alert("Failed to create Spares.");
                    }
                }
            });
        });
        
    }
    
        function validateInput(event){
            var input = event.target;
            input.value = input.value.replace(/\D/g, '')
            if (event.which < 48 || event.which > 57) {
            event.preventDefault();
          }
        }
};